packages/ui-components/src/components/UITable/UITable.tsx

```diff
@@ -126,6 +128,7 @@ export class UITable extends React.Component<UITableProps, UITableState> {
      * @param prevProps
      * @param prevState
      */
+    // eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
We should not ignore this as the cognitive complexity exists and should be fixed.
If you do not want to address this with your fix, you could make sure that you do not increase the complexity.
See https://github.com/SAP/open-ux-tools/pull/869